### PR TITLE
Create client-side navigation

### LINKED ISSUE #18: Create client side navigation framework
**The problem**
We have a basic client-side application that talks to the server. This is fine for development purposes. However, to be usable, the application needs a well thought-out client-side navigation, which is consistent from one page to another. To achieve this, we need a client-side navigation framework with a nice layout and easy navigation options.

**The solution**
We have some wireframes in this project (in the wireframes directory), that can be used when developing this client side navigation framework.
 
**The task**
Add necessary components to the client application. Add a section to the [client_app/README.md](https://github.com/oss-slu/shelter_volunteers/blob/main/client_app/README.md) file with instructions for developers on how to add a new page to the client-side.

### PR DESCRIPTION
**Pull request by loganwyas**
Fixes #18

**What was changed?**
I added a navigation bar to the client-side application, which stays present on every page. I also added some new pages (just the generic wireframes) to show the variety of pages that we will have.

**Why was it changed?**
It was changed because the client app needs to be user-friendly and deployable, so adding a navigation brings the application towards that goal.

**How was it changed?**
I added a set of React components to App.js that are associated with routing, and connected them to the pages we currently have in the application.

image

### CODE REVIEW

File: client_app/src/App.js [hunk 1]
```diff
@@ -1,14 +1,38 @@
-import {UpcomingShifts} from "./Shifts";
+import { BrowserRouter as Router, Routes, Route, Link } from "react-router-dom";
+
+import logo from "./Assets/gethelplogo.png";
+
+import { PastShifts, UpcomingShifts } from "./Shifts";
 import Shelters from "./Shelters";
+import VolunteerDashboard from "./VolunteerDashboard";
+import ShelterDashboard from "./ShelterDashboard";
```
Comment (kate-holdener) [inline client_app/src/App.js:8]: I think we don't want shelter dashboard in this app, since this is a volunteer facing app. So just volunteer dashboard.

File: client_app/src/App.js [hunk 2]
```diff
@@ -1,14 +1,38 @@
-import {UpcomingShifts} from "./Shifts";
+import { BrowserRouter as Router, Routes, Route, Link } from "react-router-dom";
+
+import logo from "./Assets/gethelplogo.png";
+
+import { PastShifts, UpcomingShifts } from "./Shifts";
 import Shelters from "./Shelters";
+import VolunteerDashboard from "./VolunteerDashboard";
```
Comment (kate-holdener) [inline client_app/src/App.js:7]: I'd put volunteer dashboard first in the navigation menu.

File: client_app/src/VolunteerDashboard.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
+function VolunteerDashboard() {
+  return (
+    <div class="volunteer-dashboard">
+      <div class="column column-1">
+        <h3>Currently Registered</h3>
```
Comment (kate-holdener) [inline client_app/src/VolunteerDashboard.js:5]: This should have a list of real upcoming shifts (use the upcoming shifts component)

File: client_app/src/App.js [hunk 4]
```diff
@@ -1,14 +1,38 @@
...
-        {Shelters()};
+      <Router>
+        <div class="main-nav">
+          <Link to="/">
+            <img class="logo" src={logo} alt="GetHelp Logo" />
+          </Link>
+          <Link to="/shelters">Shelters</Link>
+          <Link to="/past-shifts">Previous Shifts</Link>
+          <Link to="/upcoming-shifts">Upcoming Shifts</Link>
+          <Link to="/shelter-dashboard">Shelter Dashboard</Link>
```
Comment (kate-holdener) [inline client_app/src/App.js:23]:
```suggestion

```
I think we don't want to put shelter dashboard in this app, because it's a volunteer app (not shelter vacing app)

File: client_app/src/App.js [hunk 5]
```diff
@@ -1,14 +1,38 @@
...
 function App() {
   return (
     <div>
-        {UpcomingShifts()};
-        {Shelters()};
+      <Router>
+        <div class="main-nav">
+          <Link to="/">
+            <img class="logo" src={logo} alt="GetHelp Logo" />
+          </Link>
```
Comment (kate-holdener) [inline client_app/src/App.js:19]: Add a volunteer dashboard link first, which takes the user to the main page

File: client_app/src/App.js [hunk 6]
```diff
@@ -1,14 +1,38 @@
...
+          <Link to="/shelters">Shelters</Link>
```
Comment (kate-holdener) [inline client_app/src/App.js:20]: I'd reword from Shelters to "Sign up to help" or something like that

File: client_app/src/VolunteerDashboard.js [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+        </div>
+        <div class="card">
+          <h4>The Haven of Grace</h4>
+          <p>1225 Warren Street • St. Louis MO</p>
+          <p>SUN - 4:00 PM to 1:00 PM</p>
+        </div>
+      </div>
+
+      <div class="column column-2">
+        <h3>Shelters looking for Volunteers</h3>
```
Comment (kate-holdener) [inline client_app/src/VolunteerDashboard.js:24]: It would be nice if you could add the dynamic search for shelters looking for volunteers (what you currently have under shelters) here. Maybe limit it to 30 miles and display 3 shelters with the biggest need for volunteers?  Then add a button at the bottom leading the user to search for more shelter sign up options (what's currently under Shelters).

**Review by kate-holdener — CHANGES_REQUESTED** [review]
@loganwyas Thanks for getting this started! I added some inline comments. Overall, I think the volunteer dashboard should have a dynamic view of "upcoming shifts" and "shelters looking for volunteers", since we have that functionality already supported.

File: client_app/src/App.js (hunk 5 above)
Comment (loganwyas) [reply client_app/src/App.js:19]: @kate-holdener Since I currently have the volunteer dashboard as the main page, do you want me to add another link that redirects to the main page? Or is the volunteer dashboard going to be separate from the main page?

Comment (kate-holdener) [reply client_app/src/App.js:19]: @loganwyas Sorry, my comment was confusing. What I meant was: the main page should be the volunteer dashboard. The navigation menu should have an option to get back to the dashboard. 

It didn't seem right that the GetHelp logo took the user to our main page (seems like it should take the user to the GetHelp main page). Or maybe we need a different logo, which could be the link to the dashboard.

**Review by kate-holdener — APPROVED** [review]
@loganwyas I think this gives us a good start. We can fix up the look and user experience, and different look for different screen sizes later.
I pushed a change to your branch that removes check boxes from "upcoming shifts". If you are ok with that, you can push the merge button.
